File: exercises/136.single-number.py
#
# @lc app=leetcode id=136 lang=python
#
# [136] Single Number
#

# @lc code=start
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        counter = {}
        for i in nums:
            if i in counter:
                counter[i] += 1
            else:
                counter[i] = 1
        
        for i in counter:
            if counter[i] == 1:
                return i

# print(Solution.singleNumber('',[4,1,2,1,2]))  
        
# @lc code=end

File: exercises/136.single-number.ts
/*
 * @lc app=leetcode id=136 lang=typescript
 *
 * [136] Single Number
 */

// @lc code=start
function singleNumber(nums: number[]): number {
  const counter: { [key: number]: number } = {};

  for (let i of nums) {
    if (i in counter) {
      counter[i] += 1;
    } else {
      counter[i] = 1;
    }
  }

  for (let i in counter) {
    if (counter[i] === 1) {
      return Number(i);
    }
  }

  return -1;
}
// @lc code=end
